lib/heap_sort.rb

```diff
@@ -1,8 +1,57 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n log n)
+# Space Complexity: O(1)
 def heap_sort(list)
```

Comment:
Nice attempt at an in-place heapsort.  It is in place, but because heapify is recursive you need to account for the system stack in the space complexity.  This has space complexity of O(log n)